dev_guide/builds.adoc

```diff
@@ -549,189 +615,489 @@ source:
+contents of the binary archive.
+====
+
+
+[[starting-a-build]]
+
+== Starting a Build
+Manually invoke a build using the following command:
+
+----
+$ oc start-build <BuildConfigName>
+----
+
+Re-run a build using the `--from-build` flag:
+
+----
+$ oc start-build --from-build=<buildName>
+----
+
+Specify the `--follow` flag to stream the build's logs in stdout:
+
+----
+$ oc start-build <BuildConfigName> --follow
+----
+
+[[canceling-a-build]]
+
+== Canceling a Build
+Manually cancel a build using the web console, or with the following CLI command:
+
+----
+$ oc cancel-build <buildName>
+----
+
+[[accessing-build-logs]]
+
+== Accessing Build Logs
+To allow access to build logs, use one of the following commands:
+
+----
+// Stream the logs using the build directly
+$ oc logs -f build/<buildName>
+// Stream the logs of the latest build for build config
+$ oc logs -f bc/<buildConfigName>
+// Return the logs of the first build for build config
+$ oc logs --version=1 bc/<buildConfigName>
+----
+
+*Log Verbosity*
+
+To enable more verbose output, pass the `*BUILD_LOGLEVEL*` environment variable
+as part of the `*sourceStrategy*` or `*dockerStrategy*` in a `*BuildConfig*`:
+
+====
+
+----
+{
+  "sourceStrategy": {
+    ...
+    "env": [
+      {
+        "name": "BUILD_LOGLEVEL",
+        "value": "2" <1>
+      }
+    ]
+  }
+}
+----
+
+<1> Adjust this value to the desired log level.
+====
+
+NOTE: A platform administrator can set verbosity for the entire OpenShift
+instance by passing the `--loglevel` option to the `openshift start` command.
+If both `--loglevel` and `BUILD_LOGLEVEL` are specified, `BUILD_LOGLEVEL` takes precedence.
+
+Available log levels for Source builds are as follows:
+
+[horizontal]
+Level 0:: Produces output from containers running the *_assemble_* script and all encountered errors. This is the default.
+Level 1:: Produces basic information about the executed process.
+Level 2:: Produces very detailed information about the executed process.
+Level 3:: Produces very detailed information about the executed process, and a listing of the archive contents.
+Level 4:: Currently produces the same information as level 3.
+Level 5:: Produces everything mentioned on previous levels and additionally provides docker push messages.
+
+[[setting-maximum-duration]]
+== Setting Maximum Duration
+
+When defining a `*BuildConfig*`, you can define its maximum duration by setting
+the  `*completionDeadlineSeconds*` field. It is specified in seconds and is not
+set by default. When not set, there is no maximum duration enforced.
+
+The maximum duration is counted from the time when a build pod gets scheduled in
+the system, and defines how long it can be active, including the time needed to
+pull the builder image. After reaching the specified timeout, the build is
+terminated by OpenShift.
+
+The following example shows the part of a `*BuildConfig*` specifying
+`*completionDeadlineSeconds*` field for 30 minutes:
```

Comment:
Suggest "For example, the following BuildConfig fragment specifies a maximum duration of 30 minutes."